### PR TITLE
Remove stakepool operating mode.

### PR DESCRIPTION
**Pull request by jholdstock**
This completely removes the ability for dcrwallet to operate in legacy stakepool mode.

dcrstakepool was deprecated 4 years ago and we know that stakepool mode is no longer in use because we have ongoing contact with all VSP operators (who have migrated to vspd).

This PR only removes the legacy stakepool server code. It does not touch the stakepool client code, ie. the ability to purchase legacy stakepool tickets.

### Breaking changes:

1. One config flag is removed `--stakepoolcoldextkey`
1. `stakepooluserinfo` RPC is removed
1. `ticketsforaddress` RPC is removed

### CODE REVIEW

**Comment by jholdstock** [comment]
This is by far the most significant and wide-reaching change I have made to dcrwallet, so thorough review would be appreciated.

File: wallet/tickets.go [hunk 1]
```diff
@@ -10,23 +10,12 @@ import (
...
 	"github.com/jrick/bitset"
-	"golang.org/x/sync/errgroup"
 )
 
 // LiveTicketQuerier defines the functions required of a (trusted) network
 // backend that provides information about the live ticket pool.
 type LiveTicketQuerier interface {
-	//  May be removed if/when there is a header commitment to the utxo
-	//  set.
-	GetTxOut(ctx context.Context, txHash *chainhash.Hash, index uint32, tree int8, includeMempool bool) (*dcrdtypes.GetTxOutResult, error)
```
Comment (jrick) [inline wallet/tickets.go:24]: we should bump major versions for any changes to publicly exported interfaces.

File: wallet/tickets.go [hunk 2]
```diff
@@ -168,37 +91,3 @@ func (w *Wallet) LiveTicketHashes(ctx context.Context, rpc LiveTicketQuerier, in
 
 	return ticketHashes, nil
 }
-
-// TicketHashesForVotingAddress returns the hashes of all tickets with voting
-// rights delegated to votingAddr.  This function does not return the hashes of
-// pruned tickets.
-func (w *Wallet) TicketHashesForVotingAddress(ctx context.Context, votingAddr stdaddr.Address) ([]chainhash.Hash, error) {
```
Comment (jrick) [inline wallet/tickets.go:175]: although we'll bumping majors anyways for this as well.

**Comment by jrick** [comment]
This all looks great, I think all we need to do is bump our major version first.  Unfortunately that may mean a bit of conflicts after you rebase over that.

**Comment by jrick** [comment]
See #2404

**Comment by jrick** [comment]
json-rpc server version (top of internal/rpc/jsonrpc/methods.go) will need to be bumped to major version 10.

this also affects upstream json-rpc consumers who check this even though they don't use the removed method, so they will need heads up.

**Comment by jholdstock** [comment]
PR rebased onto the module version bump and RPC version bump included in the first commit.

**Comment by jrick** [comment]
Looks good, you can squash.

**Review by jrick — APPROVED** [review]

**Pull request merged**